src/Language/Pirouette/Example/IsUnity.hs

```diff
@@ -144,17 +102,15 @@ data TxInfo
 assetClassValueOf : Value -> Pair String String -> Integer
 assetClassValueOf v ac =
   match_Pair @String @String ac @Integer
-    (\(curSym : String) (tokName : String)
-     . match_Value v @Integer
-     (\(openV : KVMap String (KVMap String Integer))
-      . match_Maybe @(KVMap String Integer) (lkup @String @(KVMap String Integer) eqString openV curSym) @Integer
-          (\(tokM : KVMap String Integer)
-           . match_Maybe @Integer (lkup @String @Integer eqString tokM tokName) @Integer
-               (\(r : Integer) . r)
-               0
-          )
-          0
-    ))
+    (\(curSym : String) (tokName : String) .
+      match_Value v @Integer
+        (\(openV : KVMap String (KVMap String Integer)) .
+          match_Maybe @(KVMap String Integer) (lkup @String @(KVMap String Integer) eqString openV curSym) @Integer
```

Comment:
Looks like `match_Maybe` args order has changed. Just curious, is it just handier this way or is there a deeper reason?

Reply:
Not a very deep reason, I just try to follow how things are happening in Haskell and it looks like the prelude defines `Maybe a` as `Nothing | Just a` with `Nothing` first, so I went for that too. I also like it better personally, now that I see the difference, because usually the case of `Nothing` is fairly simple so you can read it in one go and get it out of the way and then only go read the harder case.

Reply:
Does this answer convince you or would you rather go back to how it was before? I also don't have a very strong opinion and the type-checker will have our back anyways.

Reply:
I don't think you did!